server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -6044,4 +6044,22 @@ public void testNotWarmUpSearcherInEngineCtor() throws Exception {
             }
         }
     }
+
+    public void testRefreshShouldNotFailAlreadyClosedEngine() throws Exception {
+        applyOperations(engine, generateHistoryOnReplica(between(20, 200), randomBoolean(), randomBoolean(), randomBoolean()));
+        Phaser phaser = new Phaser(2);
+        Thread refresh = new Thread(() -> {
+            phaser.arriveAndAwaitAdvance();
```

Comment:
I am not sure the additional thread is necessary here? AFAICS, if we close the engine and the do refresh, we should also get the same `AlreadyClosedException`, but we could then strenghthen the test to ensure we get the exception and do not allow refresh succeeding.